Initial app UI and responsive field monitor

Filip-Kin opened the pull request:
It'll need some more work to make the field monitor look good and have all the information it needs, but I think it's a good start.

![image](https://github.com/user-attachments/assets/849963af-36cc-4de2-9c43-4412e086d08b)

![image](https://github.com/user-attachments/assets/bcef0a7a-184e-4c8b-8d33-19559a2e576e)

Commit:
Initial UI

Commit:
Field monitor

Commit:
Responsive design

333fred commented on ui/src/fms/fms-signalr.ts:
```diff
@@ -102,6 +102,43 @@ const FieldMonitorDataMapping = {
 	pjk: 'moveToStation'
 };
 
+export enum AllianceType {
+    None = 0,
+    Red = 1,
+    Blue = 2
+}
+
+export enum StationType {
+    None = 0,
+    Station1 = 1,
+    Station2 = 2,
+    Station3 = 3
+}
+
+export enum BWUtilizationType {
+    Low = 0,
+    Medium = 1,
+    High = 2,
+    VeryHigh = 3
+}
+
+export enum MonitorStatusType {
+    Unknown = 0,
+    EStopped = 1,
+    AStopped = 2,
+    DisabledAuto = 3,
+    DisabledTeleop = 4,
+    EnabledAuto = 5,
+    EnabledTeleop = 6
+}
+
+export enum DSStationStatus {
+    Good = 0,
+    MoveStation = 1,
+    Waiting = 2,
+    Unknown = 3
+}
```
These are all defined in the openapi spec. I'd rather reuse those definitions, instead of needing to copy/paste.

Filip-Kin replied:
Yeah I was trying to use them, but since it's a declarations file you can only do type imports. I'm not sure what the best solution would be.

333fred replied:
Can we re-export them?

333fred commented on ui/bun.lockb:
Not sure that I like the idea of a binary file being checked in like this.

Filip-Kin replied:
I guess lock files aren't strictly necessary to be tracked, I'll remove it

333fred commented on ui/vite.config.ts:
```diff
@@ -2,8 +2,13 @@ import { sveltekit } from '@sveltejs/kit/vite';
 		include: ['src/**/*.{test,spec}.{js,ts}']
-	}
+    },
+    ssr: {
+        noExternal: ["svelte-hero-icons"],
+    }
```
Why is there stuff around ssr being added?

Filip-Kin replied:
It helps avoid having problems with the icons, but I can remove it since it works fine without it.

333fred commented on ui/src/app.css:
```diff
@@ -1,3 +1,35 @@
+    }
+}
+
+@media screen and (min-width: 1024px) {
+    .grid-cols-fieldmonitor {
+        grid-template-columns: minmax(auto, 1fr) 1fr 1fr 1fr auto minmax(auto, 1fr) minmax(auto, 1fr) auto;
+        width: fit-content;
+    }
+}
```
I do think that we're going to want to go with something more like the 3x2 layout I've prototyped for these types of screens in landscape mode.

Filip-Kin replied:
I'm not sure, my concern with the 3x2 layout is that it would be incorrect depending on which way the screen faces. Which, could be fixed with a button to flip the layout. We could also have a setting to switch between the traditional layout and the 3x2 layout. Options are good? Personally I prefer the traditional layout.

333fred replied:
I agree with options being good, but the main issue that I have with the vertical layout is that it doesn't make good use of the horizontal space of the screen, so you end up with dead space that could instead be used to make the display more visible across the field. The goal here is that, eventually, this would be the single interface for interacting with field statuses. If you look up in the slack channel, you can see the figma designs that I've put together on this.

333fred commented on ui/src/lib/components/Button.svelte:
Are these components copied from somewhere? Can we use some standard library for them instead of maintaining them ourselves?

Filip-Kin replied:
Tailwind ui components, for FTA Buddy I used Flowbite which is a maintained library of essentially the same components. I like the convivence of that, but I ran into issues where I wanted to change some of the components.

333fred commented on ui/src/lib/components/field-monitor/MonitorRow.svelte:
```diff
@@ -0,0 +1,121 @@
+    class="fieldmonitor-square-height md:aspect-square flex items-center justify-center font-mono text-4xl lg:text-8xl {DS_COLORS[dsStatus()]} text-black"
+    onclick={() => detailView}
+>
+    {#if dsStatus() === 2}
```
I would prefer if we used something better than just numbers here.

333fred commented on ui/src/routes/+page.svelte:
```diff
@@ -1,2 +1,150 @@
+</script>
+
+<div class="grid grid-cols-fieldmonitor lg:grid-cols-fieldmonitor-large gap-0.5 md:gap-1 mx-auto justify-center">
+    <MonitorRow monitorFrame={teams[0]} station="blue1" detailView={detailView} />
```
Would prefer if we used our AllianceType and StationType enums here, rather than having different representations of alliance and station throughout the entire app.

333fred commented:
@Filip-Kin, looks like CI is failing as well. Additionally, prettier needs to be run.

Commit:
Re-export fms api

333fred commented:
@Filip-Kin were you going to address feedback? I can update the PR later if you'd prefer.

Filip-Kin commented:
> @Filip-Kin were you going to address feedback? I can update the PR later if you'd prefer.

I will be addressing the feedback, I just haven't found the time this week to work on it.

333fred committed:
Switch to steeze-ui as per https://github.com/JustinVoitel/svelte-hero-icons/issues/37.

333fred committed:
Add defaults for team stations

333fred committed:
Run prettier, ignore the openapi file

333fred committed:
Correct MonitorRow properties

333fred committed:
Set vscode to use tabs, since that's what we have in prettier

333fred committed:
Use string types instead of numbers for readability.

333fred committed:
Remove unused decls and function

333fred committed:
Correct type errors in graph

333fred committed:
Add note page placeholder.

333fred committed:
Prettier

333fred commented:
Alright, I'm going to go ahead and merge this now.
